Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1005,6 +1005,31 @@ extension BrowserViewController: BrowserToolbarDelegate {
         tabManager.selectedTab?.reload()
     }
 
+    func browserToolbarDidLongPressReload(browserToolbar: BrowserToolbarProtocol, button: UIButton) {
+        guard #available(iOS 9.0, *) else {
+            return
+        }
+
+        guard let tab = tabManager.selectedTab where tab.webView?.URL != nil else {
+            return
+        }
+
+        let toggleActionTitle: String
+        let desktopSite: Bool
+        if let userAgent = tab.lastCustomUserAgent where !userAgent.isEmpty {
+            toggleActionTitle = NSLocalizedString("Request Mobile Site", comment: "Action Sheet Button for Requesting the Mobile Site")
+            desktopSite = false
+        } else {
+            toggleActionTitle = NSLocalizedString("Request Desktop Site", comment: "Action Sheet Button for Requesting the Desktop Site")
+            desktopSite = true
+        }
+
+        let controller = UIAlertController(title: nil, message: nil, preferredStyle: .ActionSheet)
+        controller.addAction(UIAlertAction(title: toggleActionTitle, style: .Default, handler: { action in tab.toggleDesktopSite(desktopSite) }))
+        controller.addAction(UIAlertAction(title: NSLocalizedString("Cancel", comment:"Action Sheet Cancel Button"), style: .Cancel, handler: nil))
+        presentViewController(controller, animated: true, completion: nil)
```

Comment:
We'll need to use a `UIPopoverPresentationController` here so that we don't crash on tablets. See `browserToolbarDidPressShare` in this same file for an example.